zeebe/clients/zeebe-client-spring/src/main/java/io/camunda/zeebe/spring/client/jobhandling/DefaultCommandExceptionHandlingStrategy.java

```diff
@@ -0,0 +1,94 @@
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.spring.client.jobhandling;
+
+import io.camunda.zeebe.client.api.worker.BackoffSupplier;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import java.lang.invoke.MethodHandles;
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultCommandExceptionHandlingStrategy implements CommandExceptionHandlingStrategy {
+
+  public static final Set<Status.Code> SUCCESS_CODES =
+      EnumSet.of(Status.Code.OK, Status.Code.ALREADY_EXISTS);
+  public static final Set<Status.Code> RETRIABLE_CODES =
+      EnumSet.of(
+          Status.Code.CANCELLED,
+          Status.Code.DEADLINE_EXCEEDED,
+          Status.Code.RESOURCE_EXHAUSTED,
+          Status.Code.ABORTED,
+          Status.Code.UNAVAILABLE,
+          Status.Code.DATA_LOSS);
+  public static final Set<Status.Code> IGNORABLE_FAILURE_CODES = EnumSet.of(Status.Code.NOT_FOUND);
+  public static final Set<Status.Code> FAILURE_CODES =
+      EnumSet.of(
+          Status.Code.INVALID_ARGUMENT,
+          Status.Code.PERMISSION_DENIED,
+          Status.Code.FAILED_PRECONDITION,
+          Status.Code.OUT_OF_RANGE,
+          Status.Code.UNIMPLEMENTED,
+          Status.Code.INTERNAL,
+          Status.Code.UNAUTHENTICATED);
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final BackoffSupplier backoffSupplier;
+  private final ScheduledExecutorService scheduledExecutorService;
+
+  public DefaultCommandExceptionHandlingStrategy(
+      final BackoffSupplier backoffSupplier, final ScheduledExecutorService scheduledExecutorService) {
+    this.backoffSupplier = backoffSupplier;
+    this.scheduledExecutorService = scheduledExecutorService;
+  }
+
+  @Override
+  public void handleCommandError(final CommandWrapper command, final Throwable throwable) {
+    if (StatusRuntimeException.class.isAssignableFrom(throwable.getClass())) {
+      final StatusRuntimeException exception = (StatusRuntimeException) throwable;
+      final Status.Code code = exception.getStatus().getCode();
+
+      // Success codes should not lead to an exception!
+      if (IGNORABLE_FAILURE_CODES.contains(code)) {
+        LOG.warn(
+            "Ignoring the error of type '"
+                + code
+                + "' during "
+                + command
+                + ". Job might have been canceled or already completed.");
+        // TODO: Is Ignorance really a good idea? Think of some local transaction that might need to
+        // TODO: be marked for rollback! But for sure, retry does not help at all
```

Comment:
💭  let's maybe create a general follow-up to review open TODOs

Reply:
I will create a task for that, good idea, than you.